### PR TITLE
Improve test output

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 42dcc6a1398f1a467ef0f3efab2f06b42f1288e0:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-3be87c21cdeabd1f66c245e2f2499eea6279910b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-3be87c21cdeabd1f66c245e2f2499eea6279910b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-3be87c21cdeabd1f66c245e2f2499eea6279910b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-3be87c21cdeabd1f66c245e2f2499eea6279910b-x86.apk)

File: test/spix/smoke_test.py [hunk 1]
```diff
@@ -38,41 +58,36 @@ def app():
...
             assert False  # Unexcpected exception while starting up app
 
     app.quit()
     app.quit()
     try:
-        process.wait(1)
+        process.wait(5)
     except subprocess.TimeoutExpired:
-        print("Process did not quit after 1 second. Killing it.")
+        print("Process did not quit after 5 second. Killing it.")
```
Comment (suricactus) [inline test/spix/smoke_test.py:88]:
```suggestion
        print("Process did not quit after 5 second(s). Killing it.")
```
Also having "5" as a variable `WAIT_BEFORE_FORCE_QUIT_S` would be great.

Comment (m-kuhn) [reply test/spix/smoke_test.py:88]: Already done locally

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 901960d92cfdce23a047bdfbb6f3d173822057a9:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-543979209649d2e8067ce852bf64c9687b467ffa-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-543979209649d2e8067ce852bf64c9687b467ffa-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-543979209649d2e8067ce852bf64c9687b467ffa-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-543979209649d2e8067ce852bf64c9687b467ffa-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9f56591575fef8a73a1abc905715ae66b44b11ba:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-23e8eeb53d64ba2bb425702ebd3a7afc53f34829-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-23e8eeb53d64ba2bb425702ebd3a7afc53f34829-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-23e8eeb53d64ba2bb425702ebd3a7afc53f34829-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-23e8eeb53d64ba2bb425702ebd3a7afc53f34829-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 731152ec7e53613116c09e7ddc3b1d44cbb8b4ea:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-da2e124b5ecba590d4c71e8bb674447be34a94b0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-da2e124b5ecba590d4c71e8bb674447be34a94b0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-da2e124b5ecba590d4c71e8bb674447be34a94b0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-da2e124b5ecba590d4c71e8bb674447be34a94b0-x86.apk)

File: test/spix/conftest.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        except Exception as e:
+            process_communicate(process)
+            assert False  # Unexcpected exception while starting up app
+
+    try:
+        app.quit()
+        app.quit()
+    except Exception:
+        print('Exception while trying to exit app. The process probably died.')
+        pass
```
Comment (m-kuhn) [inline test/spix/conftest.py:90]:
```suggestion
    try:
        app.quit()
        app.quit()
    except Exception:
        print('Exception while trying to exit app. The process probably died.')
        process_communicate()
        raise
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2fdc2abf659742d3484d538fb4be5e40e250c316:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-b02443d1c571faf0efbf27b5582727bb0487712e-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-b02443d1c571faf0efbf27b5582727bb0487712e-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-b02443d1c571faf0efbf27b5582727bb0487712e-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-b02443d1c571faf0efbf27b5582727bb0487712e-x86.apk)

File: test/spix/conftest.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+@pytest.fixture
+def process_communicate(process, report_summary):
+    def func():
+        try:
+            output, errors = process.communicate(timeout=0.1)
+            print(errors)
+            print(output)
+            report_summary[0].append(output.decode('utf-8'))
+            report_summary[1].append(errors.decode('utf-8'))
```
Comment (m-kuhn) [inline test/spix/conftest.py:34]:
```suggestion
            output_b, errors_b = process.communicate(timeout=0.1)
            output = output_b.decode('utf-8')
            errors = errors_b.decode('utf-8')
            print(errors)
            print(output)
            report_summary[0].append(output)
            report_summary[1].append(errors)
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d4ee71b3050070edfecb43312b7d682d2cfd84e5:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for d4ee71b3050070edfecb43312b7d682d2cfd84e5:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-cad1a5cd3b37258ea693f4c43f25ea75b462fbe0-x86.apk)

File: test/spix/smoke_test.py [hunk 4]
```diff
@@ -88,6 +36,7 @@ def test_start_app(app, screenshot_path, extra):
     )
     app.mouseClick("mainWindow/welcomeScreen/loadProjectItem_1")
     app.takeScreenshot("mainWindow", os.path.join(screenshot_path, "startup.png"))
```
Comment (m-kuhn) [inline test/spix/smoke_test.py:38]:
```suggestion
    # Crahes on CI
    # app.takeScreenshot("mainWindow", os.path.join(screenshot_path, "startup.png"))
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for db828424e6b53963ea91902d42110f6188881170:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-e592a704175256b3ed89e86bb16dd71858fa6d83-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-e592a704175256b3ed89e86bb16dd71858fa6d83-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-e592a704175256b3ed89e86bb16dd71858fa6d83-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-e592a704175256b3ed89e86bb16dd71858fa6d83-x86.apk)

**Review by nirvn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 954ee89cde633c1d11ac44bd1ced20d2846690b4:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-018994138cc8f9b40e6701f7085fab92e0285045-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-018994138cc8f9b40e6701f7085fab92e0285045-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-018994138cc8f9b40e6701f7085fab92e0285045-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1956-018994138cc8f9b40e6701f7085fab92e0285045-x86.apk)

**Pull request merged**